### PR TITLE
[AIP-141] Initial rule: num_foos -> foo_count

### PR DESCRIPTION
**Pull request by lukesneeringer**

### CODE REVIEW

File: rules/aip0141/count_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		testName  string
+		FieldName string
+		want      string
+		err       bool
+	}{
+		{"Valid", "item_count", "item_count", false},
+		{"NumItems", "num_items", "item_count", true},
+		{"NumResults", "num_results", "result_count", true},
+		{"NumErrors", "num_errors", "error_count", true},
+		{"NumMoose", "num_moose", "moose_count", true}, // singular of "moose" is "moose"
```
Comment (mingzhi) [inline rules/aip0141/count_test.go]: Add a tricky one: `{"NumTeeth", "num_teeth", "num_tooth", true}`

File: rules/aip0141/count.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	pluralize "github.com/gertd/go-pluralize"
+	"github.com/googleapis/api-linter/lint"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var count = &lint.FieldRule{
+	Name: lint.NewRuleName("core", "0141", "count-suffix"),
+	URI:  "https://aip.dev/141#guidance",
+	LintField: func(f *desc.FieldDescriptor) []lint.Problem {
+		if n := f.GetName(); strings.HasPrefix(n, "num_") {
```
Comment (mingzhi) [inline rules/aip0141/count.go]: Do we need to check if the field type is an integer: `int32`, `int64` ...?

Comment (lukesneeringer) [reply rules/aip0141/count.go]: No, I plan a different role for that.

File: rules/aip0141/count_test.go (hunk 1 above)
Comment (lukesneeringer) [reply rules/aip0141/count_test.go]: Good idea.

Comment (lukesneeringer) [reply rules/aip0141/count_test.go]: Done, but I used geese, because I am a jerk. ;-)

File: rules/aip0141/count.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+			want := pluralize.NewClient().Singular(n[4:]) + "_count"
```
Comment (mingzhi) [inline rules/aip0141/count.go]: Will people use `num_of_books` or `number_of_books` as a quantity field? For `num_of_books`, we will then produce a wired suggestion `of_book_count`.

Comment (lukesneeringer) [reply rules/aip0141/count.go]: I am not trying to address every possible contingency, and will make a separate rule later prohibiting prepositions.

Comment (mingzhi) [reply rules/aip0141/count.go]: Thanks!

**Pull request merged**